blueprint/core/blueprint_ReactApplicationRoot.h

```diff
@@ -602,13 +660,54 @@ namespace blueprint
             }, (void *) this);
         }
 
+        void registerNativeEventHandlers()
+        {
+            engine.registerNativeMethod("__BlueprintNative__", "dispatchEventToNative", [](void* stash, const juce::var::NativeFunctionArgs& args) {
+                auto self = reinterpret_cast<ReactApplicationRoot*>(stash);
+                jassert (self != nullptr);
+                jassert (args.numArguments >= 1);
+
+                const juce::String eventType = args.arguments[0].toString();
+
+                std::vector<juce::var> eventArgs;
+
+                if (args.numArguments > 1)
+                {
+                    for (auto i = 1; i < args.numArguments; ++i)
+                    {
+                        eventArgs.push_back(args.arguments[i].clone());
+                    }
+                }
+
+                std::for_each(self->eventHandlerSubscriptions.begin(),
+                              self->eventHandlerSubscriptions.end(),
+                              [=] (const EventSubscription &sub)
+                              {
+                                  if (sub.eventType == eventType)
+                                  {
+                                      auto args = juce::var::NativeFunctionArgs(
+                                                      juce::var(),
+                                                      eventArgs.data(),
```

Comment:
I think we can skip the copy-to-vector step and just pull directly from the `args` at the top, right?

```
auto args = juce::var::NativeFunctionArgs(juce::var(), args.arguments + 1, args.numArguments - 1);
```

Reply:
Yup. Good point. Ha, that was daft.